BootloaderCorePkg/Library/PciEnumerationLib/PciEnumerationLib.c

```diff
@@ -513,6 +517,20 @@ GatherDeviceInfo (
     Offset = PciParseBar (PciIoDevice, Offset, BarIndex);
   }
 
+  //
+  // Find a available PCI bar slot for PCI expansion ROM
+  //
+  if (EnumPolicy->Flag.FlagAllocRomBar == 1) {
+    BarIndex = PCI_MAX_BAR - 1;
+    while (BarIndex > 0) {
+      if (PciIoDevice->PciBar[BarIndex].Length == PciBarTypeUnknown) {
```

Comment:
do you mean PciIoDevice->PciBar[BarIndex].BarType when comparing with PciBarTypeUnknown?

Reply:
Ahh, yes, you are right.
Let me fix it.